### ISSUE TITLE
pull: Unclear error message when remote support is not installed

### ISSUE THREAD

**Issue by nielstenboom** [iterative/dvc#6870]
# Bug Report

## Description

If you've installed DVC via pip without the remote support you need (for example S3), the error message is very unclear.

### Reproduce

1. Setup fresh repo & `pip install dvc[s3]`
2. Copy dataset.zip to repo
3. `dvc init` & `dvc add dataset.zip`
4. Setup remote & `dvc push`
5. Move to fresh python environment
6. `pip install dvc`
7. Clone repo 
8. `dvc pull -v`

### Expected

I would've expected an error message saying something like: "the remote type (s3) is not supported in your dvc installation". 

Instead you get the following output:

```
$ dvc pull -v
2021-10-26 10:35:07,698 DEBUG: failed to pull cache for 'models'      
2021-10-26 10:35:07,699 DEBUG: failed to pull cache for 'data/evaluation'
2021-10-26 10:35:08,269 WARNING: No file hash info found for 'data/evaluation'. It won't be created.                                                                                                                                            
2021-10-26 10:35:08,304 WARNING: No file hash info found for 'models'. It won't be created.                                                                                                                                                     
2 files failed                                                                                                                                                                                                                                  
2021-10-26 10:35:08,326 ERROR: failed to pull data from the cloud - Checkout failed for following targets:
data/evaluation
models
Is your cache up to date?
<https://error.dvc.org/missing-files>
```

Took me a couple hours to figure it out I had to reinstall dvc with s3 support enabled 😅 

### Environment information

<!--
This is required to ensure that we can reproduce the bug.
-->

**Output of `dvc doctor`:**

```console
$ dvc doctor

DVC version: 2.8.2 (pip)
---------------------------------
Platform: Python 3.9.7 on Linux-4.14.246-187.474.amzn2.x86_64-x86_64-with-glibc2.31
Supports:
        webhdfs (fsspec = 2021.10.1),
        http (aiohttp = 3.7.4.post0, aiohttp-retry = 2.4.5),
        https (aiohttp = 3.7.4.post0, aiohttp-retry = 2.4.5),
Cache types: hardlink, symlink
Cache directory: nfs4 on xxxxxx.efs.eu-west-1.amazonaws.com:/data-pvc
Caches: local
Remotes: s3
Workspace directory: nfs4 on xxxxx.efs.eu-west-1.amazonaws.com:/data-pvc
Repo: dvc, git
```

**Comment by nsorros** [comment]
I faced the same error and was able to reproduce it even without the cache change. My steps are

```
mkdir issue
cd issue
git init
dvc init
mkdir data
touch data/data.csv
dvc add data
dvc remote add remote s3://bucket
dvc push
```

then to replicate the problem
```
cd ..
mkdir test_issue
cd test_issue
git clone ../issue
python -m venv venv
source venv/bin/activate
pip install dvc
dvc pull
```

**Comment by pared** [comment]
@nsorros what is the result for `dvc version` for the environment that produces wrong error message?

**Comment by efiop** [comment]
@nsorros Please post full verbose log for `dvc pull -v` and post `dvc doctor` output.

**Comment by nsorros** [comment]
> @nsorros Please post full verbose log for `dvc pull -v` and post `dvc doctor` output.

The error I was getting will be addressed by #7501
